packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,262 @@
+# Checkbox Decisions <!-- omit in toc -->
+
+- [Design](#design)
+- API
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=13227%3A163026) to all variants of the Checkbox component
+
+## Anatomy
+
+- **Checkbox**
+  - **CheckboxInput** - Hidden native <input \/> element
+  - **CheckboxLabel** - The Checkbox's <label \/> element
+  - **CheckboxLabelText** - The cross-platform Text component
+  - **CheckboxHelperText** - The helper text component
+- **CheckboxGroup**
+  - **CheckboxGroupFieldset** - Would render a native <fieldset \/>
+  - **CheckboxGroupLegend** - Would render a native <legend \/>
+  - **CheckboxGroupHelperText** - The helper text component
+
+Checkbox Component's Internal Anatomy:
+
+```tsx
+const Checkbox = () => {
+  return (
+    <>
+      <CheckboxLabel>
+        <VisuallyHidden>
+          <CheckboxInput />
+        </VisuallyHidden>
+        <CheckboxLabelText />
+      </CheckboxLabel>
+      <CheckboxHelperText />
+    </>
+  );
+};
+```
+
+CheckboxGroup Component's Internal Anatomy:
+
+```tsx
+const CheckboxGrop = () => {
+  return (
+    <CheckboxGroupFieldset>
+      <CheckboxGroupLegend>title</CheckboxGroupLegend>
+      {children}
+      <CheckboxGroupHelperText type="default | negative" />
+    </CheckboxGroupFieldset>
+  );
+};
+```
+
+## `Checkbox` API
+
+| Prop                | Type       | Required | Default   | Description                                                                                                                                              |
+| ------------------- | ---------- | -------- | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| children            | `string`   | No       | undefined | The text to be rendered as CheckboxLabel                                                                                                                 |
+| helperText          | `string`   | No       | undefined | The helper text to be rendered                                                                                                                           |
+| isChecked           | `boolean`  | No       | false     | state of the checkbox, if explicitly set the checkbox will be controlled                                                                                 |
+| defaultChecked      | `boolean`  | No       | false     | default state of the checkbox, if isChecked is not provided the checkbox will be uncontrolled                                                            |
```

Comment:
`defaultChecked` will act as an initializer for when we want the checkbox to be uncontrolled. If consumer passes `isChecked={true}` it will _always_ remain checked even when a user tries to toggle the checkbox until consumer changes it to `isChecked={false}` but in case of `defaultChecked={true}` it will start at an initial state of being checked but when a user clicks on the checkbox, it will get unchecked even though `defaultChecked` remains true.